Plugins/Public/base_plugin/Main.cpp

```diff
@@ -71,14 +61,30 @@ list<REPAIR_ITEM> set_base_repair_items;
 map<uint, uint> set_base_crew_consumption_items;
 map<uint, uint> set_base_crew_food_items;
 
+uint set_crew_check_frequency = 43200;
+
 /// The commodity used as crew for the base
 uint set_base_crew_type;
 
+unordered_set<uint> humanCargoList;
+
 /// A return code to indicate to FLHook if we want the hook processing to continue.
 PLUGIN_RETURNCODE returncode;
 
-/// Map of item nickname hash to recipes to construct item.
-map<uint, RECIPE> recipes;
+/// Global recipe map
+map<uint, RECIPE> recipeMap;
+
+/// Maps of shortcut numbers to recipes to construct item.
+map<wstring, map<uint, RECIPE>> recipeCraftTypeNumberMap;
+map<wstring, map<wstring, RECIPE>> recipeCraftTypeNameMap;
+map<uint, vector<wstring>> factoryNicknameToCraftTypeMap;
+map<wstring, RECIPE> moduleNameRecipeMap;
+map<uint, RECIPE> moduleNumberRecipeMap;
+map<wstring, map<uint, RECIPE>> craftListNumberModuleMap;
+set<wstring> buildingCraftLists;
+
+void AddFactoryRecipeToMaps(const RECIPE& recipe);
```

Comment:
we're just going to leave these functions here defined unimplemented near the top of a cpp file?

Reply:
Simply a c++ forward declaration so I don't need to clutter top of the file with the method and can stash it at the bottom.